### PR TITLE
Update roadmap to add auth features

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

File: _docs/reference/release-roadmap.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Istio 0.2 will be our second release, and will work across multiple Kubernetes c
 - Support for VMs joining an Istio Mesh.
 - Fully-automated injection of Envoy proxies into Kubernetes pods.
 - Standard extension points for addons and plugins.
+- Expanded service-to-service authentication support for VMs.
```
Comment (smawson) [inline _docs/reference/release-roadmap.md]: We already have "Support for VMs joining an Istio Mesh". What does this add?

That item already includes service naming, authn, authz, ....

File: _docs/reference/release-roadmap.md [hunk 2]
```diff
@@ -32,6 +33,7 @@ Istio 0.3 will be our third release, focused on performance, scale, and stabilit
 - Client-oriented telemetry collection and distribution.
 - Global load balancing with autoscaling.
 - Support for the Open Service Broker APIs.
+- Authentication support including pluggable CA, container identity, etc.
```
Comment (smawson) [inline _docs/reference/release-roadmap.md]: How about:

Support for pluggable Authentication providers.

Or something like that.

File: _docs/reference/release-roadmap.md (hunk 1 above)
Comment (linsun) [inline _docs/reference/release-roadmap.md]: How is this new auth feature different from "Expanded authorization with callouts to external services.", copied from https://istio.io/docs/reference/release-roadmap.html?

Also, do we have any example on users can explore this pluggable auth support?

Comment (wenchenglu) [inline _docs/reference/release-roadmap.md]: "Expanded authorization with callouts to external services." is for integration with external authorization system. This item is about adding authentication support for services running on VMs (non-k8s environment). As Sven said, authn is already included as part of "Support for VMs joining an Istio Mesh" item above. We may leave this as is, or update this item to include clarify which features are included

File: _docs/reference/release-roadmap.md (hunk 2 above)
Comment (wenchenglu) [inline _docs/reference/release-roadmap.md]: How about: 
Pluggable identity (key/cert) provisioning system that allow plugging in custom CA (Vault) and custom-defined identity.

File: _docs/reference/release-roadmap.md (hunk 1 above)
Comment (wattli) [inline _docs/reference/release-roadmap.md]: @linsun, the one I added is for authentication, not authorization. 

Revert as Sven and Wencheng suggested.

File: _docs/reference/release-roadmap.md (hunk 2 above)
Comment (wattli) [inline _docs/reference/release-roadmap.md]: Mostly follow the suggestion, except for making it a little concise.

File: _docs/reference/release-roadmap.md (hunk 1 above)
Comment (linsun) [inline _docs/reference/release-roadmap.md]: @wattli thanks for clarifying that!

**Pull request merged**